From bbbb4313846efaba9546e86052940f0a79b8948f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B6rn=20Kottmann?= <joern@apache.org>
Date: Wed, 26 Apr 2017 10:46:48 +0200
Subject: [PATCH] OPENNLP-1046: Correctly join tokens to text string

The text was one space too long which results in a different
parse tree if the method is used to reproduce an existing
parse tree as it is done by the parser evaluation tool.
---
 .../main/java/opennlp/tools/cmdline/parser/ParserTool.java | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/opennlp-tools/src/main/java/opennlp/tools/cmdline/parser/ParserTool.java b/opennlp-tools/src/main/java/opennlp/tools/cmdline/parser/ParserTool.java
index 499fa5810..d8d390268 100644
--- a/opennlp-tools/src/main/java/opennlp/tools/cmdline/parser/ParserTool.java
+++ b/opennlp-tools/src/main/java/opennlp/tools/cmdline/parser/ParserTool.java
@@ -72,11 +72,8 @@ public static Parse[] parseLine(String line, Parser parser, Tokenizer tokenizer,
 
     // tokenize
     List<String> tokens = Arrays.asList( tokenizer.tokenize(line));
-    StringBuilder sb = new StringBuilder();
-    for (String tok : tokens) {
-      sb.append(tok).append(" ");
-    }
-    String text = sb.substring(0, sb.length());
+    String text = String.join(" ", tokens);
+
     Parse p = new Parse(text, new Span(0, text.length()), AbstractBottomUpParser.INC_NODE, 0, 0);
     int start = 0;
     int i = 0;
